### PR TITLE
[Excel] (Custom functions) Add streaming spill array snippet

### PR DESCRIPTION
**Pull request by alison-mk**
[Internal work item](https://office.visualstudio.com/OC/_workitems/edit/6057195).

Addressing customer issue #3411.

### CODE REVIEW

File: docs/excel/custom-functions-dynamic-arrays.md [hunk 1]
```diff
@@ -58,6 +60,33 @@ function spillRectangle() {
...
+      let thirdResult = 2;
+
+      const timer = setInterval(() => {
+        firstResult += amount;
+        secondResult += amount;
+        thirdResult += amount;
+        invocation.setResult([[firstResult], [secondResult], [thirdResult]]);
+      }, 1000);
+
+      invocation.onCanceled = () => {
```
Comment (davidchesnut) [inline docs/excel/custom-functions-dynamic-arrays.md:84]: This requires the @cancelable attribute in the code section above. I'm not sure how to test this though to see if it really cancels.

File: docs/excel/custom-functions-dynamic-arrays.md [hunk 2]
```diff
@@ -58,6 +60,33 @@ function spillRectangle() {
 }
 ```
 
+The fourth example shows how to return a dynamic spill array from a streaming function. The results spill down, like the first example, and increment once a second based on the `amount` parameter. To learn more about streaming functions, see [Make a streaming function](custom-functions-web-reqs.md#make-a-streaming-function).
+
+```javascript
+    /** @CustomFunction
```
Comment (davidchesnut) [inline docs/excel/custom-functions-dynamic-arrays.md:66]:
```suggestion
    /**
 * Increment the cells with a given amount every second. Creates a dynamic spilled array with multiple results
 * @customfunction
 * @param {number} amount The amount to add to the cell value on each increment.
 * @param {CustomFunctions.StreamingInvocation<number[][]>} invocation Parameter to send results to Excel or respond to the user canceling the function. A dynamic array.
 * 
 */
```

Please use this comment section instead for correct case and layout. Also there is no return value, so I moved the description to the top. This makes easier to paste into Script Lab. :-)

File: docs/excel/custom-functions-web-reqs.md [hunk 3]
```diff
@@ -127,6 +126,9 @@ function increment(incrementBy, invocation) {
 }
 ```
 
+> [!NOTE]
```
Comment (davidchesnut) [inline docs/excel/custom-functions-web-reqs.md:129]: Later in this article is a note that says all cancelable functions must return 1 value. This sample returns void. So I'm not sure if the note needs an update.

**Review by davidchesnut — CHANGES_REQUESTED** [review]
Nice. I added a few suggestions to take a look at. Thanks!

File: docs/excel/custom-functions-dynamic-arrays.md (hunk 1 above)
Comment (alison-mk) [reply docs/excel/custom-functions-dynamic-arrays.md:90]: My understanding of our documentation is that @cancelable is not relevant for streaming functions: 
image

Screenshot is from this article section: https://learn.microsoft.com/en-us/office/dev/add-ins/excel/custom-functions-web-reqs#cancel-a-function

Comment (alison-mk) [reply docs/excel/custom-functions-dynamic-arrays.md:90]: I based this new streaming code sample off the streaming sample in this article: https://learn.microsoft.com/en-us/office/dev/add-ins/excel/custom-functions-web-reqs#make-a-streaming-function, which has an onCanceled method and doesn't include @cancelable in the JS doc. Let me know if the existing snippet needs to be updated, too

File: docs/excel/custom-functions-web-reqs.md (hunk 3 above)
Comment (alison-mk) [reply docs/excel/custom-functions-web-reqs.md:129]: It seems like we each interpret this note differently. Perhaps we can sync offline and then adjust the note accordingly.

**Comment by alison-mk** [comment]
@davidchesnut Thanks for the review! Let's sync offline to discuss further.

File: docs/excel/custom-functions-dynamic-arrays.md (hunk 1 above)
Comment (alison-mk) [reply docs/excel/custom-functions-dynamic-arrays.md]: I clarified this note based on our offline conversation and using information from https://learn.microsoft.com/en-us/office/dev/add-ins/excel/custom-functions-json-autogeneration#cancelable

Ready for you to take another look. Thanks!

**Review by davidchesnut — APPROVED** [review]
Looks great!

**Pull request merged**